api/src/services/spatial-service.ts

```diff
@@ -221,24 +221,17 @@ export class SpatialService extends DBService {
    * @return {*}  {Promise<ISubmissionSpatialComponent[]>}
    * @memberof SpatialService
    */
-  async findSpatialMetadataBySubmissionSpatialComponentId(
-    submissionSpatialComponentId: number
-  ): Promise<Record<string, string>> {
+  async findSpatialMetadataBySubmissionSpatialComponentIds(
+    submissionSpatialComponentIds: number[]
+  ): Promise<Array<Record<string, any>>> {
     const userService = new UserService(this.connection);
 
-    if (await userService.isSystemUserAdmin()) {
-      const adminResponse = await this.spatialRepository.findSpatialMetadataBySubmissionSpatialComponentIdasAdmin(
-        submissionSpatialComponentId
-      );
-      return (
-        (adminResponse?.spatial_component?.spatial_data?.features?.[0]?.properties as Record<string, string>) || {}
-      );
-    }
-
-    const response = await this.spatialRepository.findSpatialMetadataBySubmissionSpatialComponentId(
-      submissionSpatialComponentId
-    );
+    const response = (await userService.isSystemUserAdmin())
+      ? this.spatialRepository.findSpatialMetadataBySubmissionSpatialComponentIdsAsAdmin(submissionSpatialComponentIds)
+      : this.spatialRepository.findSpatialMetadataBySubmissionSpatialComponentIds(submissionSpatialComponentIds);
 
-    return (response?.spatial_component?.spatial_data?.features?.[0]?.properties as Record<string, string>) || {};
+    return (await response)
+      .map((row) => row.spatial_component_properties)
+      .filter((row): row is Record<string, any> => Boolean(row));
```

Comment:
Rather than cast this to a Record, we should just change the response of this function to an array of  `GeoJsonProperties`, to match what `spatial_component_properties` is already set to.

Reply:
My reasoning was that `GeoJsonProperties` is basically just `Record<string, any> | null`, so I cast it as a `Record` so it would never be `null`. Although we have a check for `spatial_component != {}` in the SQL query, I don't think this necessitates that the `spatial_component_properties` value can't be `null` (but I might be wrong?) - hence I added the `filter` call. If we go for changing the service response to `GeoJsonProperties[]`, we should probably do the same for the API hook in the frontend. What do you think?

Reply:
Yeah I was thinking about the `null` aspect too.  It is technically correct that a geojson feature can have null properties, even if 
we don't think we will ever set it to null in our database.

But on the other hand, if we think that null is unrealistic since we control the data in those tables, then the other option is to just define the properties as `Record<string, any>` and not bother using `GeoJsonProperties` at all.

I think it would be better to either use `GeoJsonProperties` and then be forced to handle a few extra null checks at the app level, or use `Record<string, any>` and not worry about null checks, and have it blow up if null gets returned unexpectedly.  But what we don't want to do is filter out nulls silently, which we don't really think should exist anyways, and then be left unaware of unexpected nulls in the database.

I guess I would lean towards using `GeoJsonProperties` if I had to pick, only because its the most correct, even if it means we have to do some extra checks in the frontend (which I think will be fairly minimal anyways, so not a big deal).

And yeah, whichever option we pick, we should propagate it everywhere so that the types at all levels are accurate.

Reply:
> But what we don't want to do is filter out nulls silently, which we don't really think should exist anyways, and then be left unaware of unexpected nulls in the database.

Well put.

> I think it would be better to either use GeoJsonProperties and then be forced to handle a few extra null checks at the app level...

I think I will opt for this.